### PR TITLE
Convert raw SQL in Spree::Zone.with_member_ids to Arel

### PR DESCRIPTION
**Pull request by mamhoff**
Makes the code look nicer and indentation easier.

### CODE REVIEW

**Comment by tvdeyen** [comment]
Two in a row? @jhawthorn and @mamhoff you are crazy! 

Nice :+1:

File: core/app/models/spree/zone.rb [hunk 1]
```diff
@@ -15,17 +15,19 @@ class Zone < Spree::Base
...
-              "(spree_zone_members.zoneable_type = 'Spree::State' AND
-                 spree_zone_members.zoneable_id IN (?))
-               OR (spree_zone_members.zoneable_type = 'Spree::Country' AND
-                 spree_zone_members.zoneable_id IN (?))",
-              state_ids,
-              country_ids
-            ).uniq
-          end
+    scope :with_member_ids, ->(state_ids, country_ids) do
+      return none unless state_ids || country_ids
```
Comment (jhawthorn) [inline core/app/models/spree/zone.rb]: maybe this should check for `presence?` if `state_ids == []` and `country_ids == []` we know there will be no results

**Comment by jhawthorn** [comment]
With our without suggestion :+1:

**Comment by cbrunsdon** [comment]
:+1:, also with our without hawth's suggestion

**Comment by mamhoff** [comment]
Suggestion added, and made the condition for the early return a little less terse.

**Comment by Senjai** [comment]
:shipit: :+1:

**Comment by jhawthorn** [comment]
:+1:

**Pull request merged**